docs/integration/driver/display/lcd_stm32_guide.rst

```diff
@@ -0,0 +1,320 @@
+------------
+
+This guide is intended to be a step-by-step instruction of how to configure the STM32Cube HAL with the new TFT-LCD display drivers introduced in LVGL v9.0. The example code has been tested on the STM32F746-based Nucleo-F746ZG board with an ST7789-based LCD panel connected via SPI. The application itself and the hardware configuration code were generated with the STM32CubeIDE 1.14.0 tool.
+
+.. tip::
+	ST Micro provide their own TFT-LCD drivers in their X-CUBE-DISPLAY Software Extension Package. While these drivers can be used with LVGL as well, the LVGL LCD drivers do not depend on this package.
+	
+	The LVGL LCD drivers are meant as an alternative, simple to use API to implement LCD support for your LVGL-based project on any platform. Moreover, even in the initial release we support more LCD controllers than X-CUBE-DISPLAY currently provides, and we plan to add support for even more LCD controllers in the future.
+	
+	Please note however, that – unlike X-CUBE-DISPLAY – the LVGL LCD drivers do not implement the communication part, whether SPI, parallel i8080 bus or other. It is the user's responsibility to implement – and optimize – these on their chosen platform. LVGL will only provide examples for the most popular platforms.
+
+By following the steps you will have a fully functional program, which can be used as the foundation of your own LVGL-based project. If you are in a hurry and not interested in the details, you can find the final project `here <https://github.com/lvgl/lv_port_lcd_stm32>`__. You will only need to configure LVGL to use the driver corresponding to your hardware (if it is other than the ST7789), and implement the function ``ui_init()`` to create your widgets.
+
+.. note::
+
+	This example is not meant as the best possible implementation, or the recommended solution. It relies solely on the HAL drivers provided by ST Micro, which favor portability over performance. Despite of this the performance is very good, thanks to the efficient, DMA-based implementation of the drivers.
+
+.. note::
+
+	Although the example uses FreeRTOS, this is not a strict requirement with the LVGL LCD display drivers.
+	
+You can find the source code snippets of this guide in the `lv_port_lcd_stm32_template.c <https://github.com/lvgl/lvgl/examples/porting/lv_port_lcd_stm32_template.c>`__ example.
+
+Hardware configuration
+----------------------
+
+In this example we'll use the SPI1 peripheral to connect the microcontroller to the LCD panel. Besides the hardware-controlled SPI pins SCK and MOSI we need some additional output pins for the chip select, command/data select, and LCD reset:
+
+==== ============= ======= ==========
+pin  configuration LCD     user label
+==== ============= ======= ==========
+PA4  GPIO_Output   CS	   LCD_CS
+PA5  SPI1_SCK	   SCK	   --
+PA7  SPI1_MOSI	   SDI     --
+PA15 GPIO_Output   RESET   LCD_RESET
+PB10 GPIO_Output   DC      LCD_DCX
+==== ============= ======= ==========
+
+Step-by-step instructions
```

Comment:
Do we really need such a detailed guide here? 

I think we should explain only the LVGL related things and not general topic (such as clock config). For such a detailed and specific configuration we should have a blog post or YouTube video instead that we can link from here.

I suggest updating [this file](https://docs.lvgl.io/master/integration/chip/stm32.html) instead where we explain:
1. How to integrate LVGL
   - clone LVGL
   - lv_tick 
2. STM devices can be used with 
   - LVGL's built in display controller drivers
      - 1-2 paragraph of description about what to configure in Cube IDE (note that STm devices can be used with outher IDEs as well)  
      - link to `lv_port_stm32_lcd`
   - X-CUBE-DISPLAY (just one paragraph)
   - LTDC: 1-2 paragraphs. 

I hope we can have an official collaboration with ST in the future. However, until that I would like focus on our Official partners. 

For example it would be nice to have the same description for ESP as well.

Reply:
I think it is very useful to provide a step-by-step guide like this, because the difficult part is not how to integrate LVGL, but how to configure the hardware. While this could be a blog post or YouTube video, including it in the LVGL documentation will immediately help novice users with their LVGL project, saving the hassle of searching the Internet for solutions (which may be outdated and may not be correct). Having this guide would have saved me days of work... :-)

The reason for making the guide so detailed is that leaving out any of these steps (e.g., clock or DMA configuration) will make LVGL perform sub-optimally. And that would cause user complaints of LVGL in general or discarding LVGL alltogether.

Reply:
Okay let's leave it as it is but please add link to it [here](
https://docs.lvgl.io/master/integration/chip/stm32.html)